source/intel.js

```diff
@@ -100,10 +100,50 @@ const display_tech_trading = () => {
 	}
 	return tech_trade_screen
 }
+let cached_events = [];
+
+const update_event_cache = (fetchSize) => {
+		jQuery.ajax({
+			type: 'POST',
+			url: "/trequest/fetch_game_messages",
+			async: true,
+			data: {
+				type: "fetch_game_messages",
+				count: cached_events.length > 0 ? fetchSize : 100000,
+				offset: 0,
+				group: "game_event",
+				version: NeptunesPride.version,
+				game_number: NeptunesPride.gameNumber
+			},
+			success: recieve_new_messages,
+			error: console.log,
+			dataType: "json"
+		});
+}
 const recieve_new_messages = (response) => {
 	const npui = NeptunesPride.npui
 	const universe = NeptunesPride.universe
-	const players = get_ledger(response.report.messages)
```

Comment:
Note to self while reading: 

`get_ledger` would return a list of dictoinary elements of `{ "player_name": debt }`.